src/librustc/hir/lowering.rs

```diff
@@ -928,13 +928,20 @@ impl<'a> LoweringContext<'a> {
                 hir::ItemDefaultImpl(self.lower_unsafety(unsafety),
                                      self.lower_trait_ref(trait_ref))
             }
-            ItemKind::Impl(unsafety, polarity, ref generics, ref ifce, ref ty, ref impl_items) => {
+            ItemKind::Impl(unsafety,
+                           polarity,
+                           defaultness,
+                           ref generics,
+                           ref ifce,
+                           ref ty,
+                           ref impl_items) => {
                 let new_impl_items = impl_items.iter()
                                                .map(|item| self.lower_impl_item_ref(item))
                                                .collect();
                 let ifce = ifce.as_ref().map(|trait_ref| self.lower_trait_ref(trait_ref));
                 hir::ItemImpl(self.lower_unsafety(unsafety),
                               self.lower_impl_polarity(polarity),
+                              self.lower_defaultness(defaultness, true /* [1] */),
```

Comment:
Originally I've copied the `lower_defaultness` method from line [1428](https://github.com/rust-lang/rust/blob/master/src/librustc/hir/lowering.rs#L1428) 
where there is a footnote explaining the reason for setting `true` in the `has_value` field of `hir::Defaultness`. 
In my case the `has_value` method is never called in an `impl`'s `defaultness` but i'll leave it as always `true` in order to not cause an assertion failure inside the `lower_defaultness` function.
I'm putting a footnote to explain this.